fish-rust/src/common.rs

```diff
@@ -1634,6 +1634,13 @@ pub fn restore_term_foreground_process_group_for_exit() {
     }
 }
 
+fn slice_contains_slice<T: Eq>(a: &[T], b: &[T]) -> bool {
+    a.windows(b.len()).any(|aw| aw == b)
+}
+
+#[cfg(target_os = "linux")]
+static IS_WINDOWS_SUBSYSTEM_FOR_LINUX: once_cell::race::OnceBool = once_cell::race::OnceBool::new();
```

Comment:
Right, I always use `once_cell:sync` even if it's not necessary, I should recheck all of them

Reply:
I think any use of `Lazy<Mutex<...>>` is suspect since Lazy does locking for you. I see an instance of that in common.rs for shell modes.